src/snapred/resources/application.yml

```diff
@@ -137,6 +136,11 @@ logging:
 samples:
   home: ${instrument.calibration.sample.home}
 
+version:
+  error: "uninitialized"  # alphanumeric
+  default: 0 # alphanumeric
```

Comment:
Version is always an internally represented as an `int`.  The user can choose to have it represented by a string through this setting.  Internally, the default version is `-1`.

If a version wishes, this could be set to `default: "default"`, for instance, and the default version will be named "default" instead of `0`.

Reply:
to clear up possible confusion, we could nest these values under a "friendlyName" heading?
i.e. `version.friendlyName,error`